torchrl/collectors/collectors.py

```diff
@@ -1595,7 +1606,19 @@ def _queue_len(self) -> int:
         raise NotImplementedError
 
     def _run_processes(self) -> None:
+        if self.num_threads is None:
+            import torchrl
+
+            total_workers = self._total_workers_from_env(self.create_env_fn)
+            self.num_threads = max(
+                1, torchrl._THREAD_POOL - total_workers
+            )  # 1 more thread for this proc
+
         torch.set_num_threads(self.num_threads)
+        assert torch.get_num_threads() == self.num_threads
```

Comment:
@vmoens Not sure this assert is wanted here. It will probably break with on OSX where just doing this breaks:

```python
print(torch.get_num_threads())
    torch.set_num_threads(5)
    print(torch.get_num_threads())
    torch.set_num_threads(1)
    print(torch.get_num_threads())

```
gives
```bash
8
5
/Users/moalla/projects/claire/po-dynamics/tests/issue_num_threads_collector.py:27: UserWarning: Cannot set number of intraop threads after parallel work has started or after set_num_threads call when using native parallel backend (Triggered internally at /Users/runner/work/_temp/anaconda/conda-bld/pytorch_1702400234613/work/aten/src/ATen/ParallelNative.cpp:230.)
  torch.set_num_threads(1)
5
```